### ISSUE TITLE
Tweets appear out of order

### ISSUE THREAD

**Issue by TheJhyde** [Tweechable/tweechable_final#69]
If you edit a tweet in preexisting lesson, the app will display that tweet at the bottom of the list of tweets, disrupting the order you probably wanted. There's also no way to insert a tweet in the middle of a set of tweets.

We need a way to store the proper order of tweets in the back end and a way to edit the order to tweets on the front end.

**Comment by vkoves** [comment]
I've moved this into the milestone since it disrupts how people work with Tweechable, and would add a requirement for training for people not to make mistakes with this. Also it can be very simply fixes by adding an `index` field of type `integer` to the tweets, then making lessons ordered by the index of each tweet. Ideally I'd like a drag and drop reorder system for this, but for MVP this should just be an editable index field.

**Comment by McEileen** [comment]
Hi, I'm looking at this issue now much, much later, and our current schema has a field on the `tweet` model, `twitter_id`, which I think fulfills the role of an index.  See below:

  ```
create_table "tweets", force: :cascade do |t|
    t.integer  "lesson_id"
    t.string   "text"
    t.string   "cited_src"
    t.boolean  "approved",             default: true
    t.integer  "twitter_id", limit: 8
    t.datetime "created_at"
    t.datetime "updated_at"
  end

  add_index "tweets", ["lesson_id"], name: "index_tweets_on_lesson_id", using: :btree
  add_index "tweets", ["twitter_id"], name: "index_tweets_on_twitter_id", using: :btree
```
 I especially think `twitter_id` is meant to serve as an index for tweets because of the final line that adds an index.  I'm going to work on this issue and will write accompanying tests.

**Comment by vkoves** [comment]
I believe Twitter ID is the ID of the tweet so it can be deleted later. We
thought this at first too but looking into it suggested otherwise, based on
when it was called.

On May 8, 2017 11:01 PM, "McEileen" <notifications@github.com> wrote:

Hi, I'm looking at this issue now much, much later, and our current schema
has a field on the tweet model, twitter_id, which I think fulfills the role
of an index. See below:

create_table "tweets", force: :cascade do |t|
  t.integer  "lesson_id"
  t.string   "text"
  t.string   "cited_src"
  t.boolean  "approved",             default: true
  t.integer  "twitter_id", limit: 8
  t.datetime "created_at"
  t.datetime "updated_at"
end

add_index "tweets", ["lesson_id"], name: "index_tweets_on_lesson_id",
using: :btree
add_index "tweets", ["twitter_id"], name:
"index_tweets_on_twitter_id", using: :btree

I especially think twitter_id is meant to serve as an index for tweets
because of the final line that adds an index. I'm going to work on this
issue and will write accompanying tests.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/Tweechable/tweechable_final/issues/69#issuecomment-300055282>,
or mute the thread

**Comment by McEileen** [comment]
Ah, thanks!  (If you're available, could you let me know what you think about issue #113?)

**Comment by McEileen** [comment]
I am closing this issue because I resolved it with PR #114.
